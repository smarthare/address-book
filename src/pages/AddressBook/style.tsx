import { styled } from "@mui/material/styles";

export const Container = styled("div")(() => ({
  background: "#f6f5ea",
  flexGrow: 1,
  padding: "3.5vw 4vw",

  "& h1": {
    fontSize: "2.25rem",
    fontFamily: "Gotham",
    fontWeight: "bold",
    margin: 0,
  },

  "& p": {
    margin: 0,
    fontSize: "1.125rem",
    fontFamily: "Gotham",
    lineHeight: 1.5,
  },
}));
