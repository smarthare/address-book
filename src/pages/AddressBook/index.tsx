import { useContext, useEffect, useState } from "react";
import { Grid, Typography } from "@mui/material";

import SelectAddressDialog from "components/Dialogs/SelectAddressDialog";
import AutoAddDialog from "components/Dialogs/AutoAddDialog";
import ManualAddDialog from "components/Dialogs/ManualAddDialog";
import { BequestButton, BequestInput } from "components/Bequest";

import { AddressContext } from "contexts/AddressContext";
import concatAddress from "utils/concatAddress";
import { Address } from "types/address";
import { AddressInputWrapper, Container } from "./style";

const AddressBook = () => {
  const { currentId, addrs } = useContext(AddressContext);

  const [openSelect, setOpenSelect] = useState<boolean>(false);
  const [openAuto, setOpenAuto] = useState<boolean>(false);
  const [openManual, setOpenManual] = useState<boolean>(false);
  const [currentAddr, setCurrentAddr] = useState<Address>();

  useEffect(() => {
    const addr = addrs.find((addr) => addr.id === currentId);
    addr && setCurrentAddr(addr);
  }, [currentId, addrs]);

  // handling SelectAddressDialog
  const handleOpenSelect = () => {
    setOpenSelect(true);
  };
  const handleCloseSelect = () => {
    setOpenSelect(false);
  };

  // handling AutoAddAddressDialog
  const handleOpenAuto = () => {
    setOpenManual(false);
    setOpenSelect(false);
    setOpenAuto(true);
  };
  const handleCloseAuto = (flg?: boolean) => {
    setOpenAuto(false);
    !flg && setOpenSelect(true);
  };

  // handling ManualAddAddressDialog
  const handleOpenManual = () => {
    setOpenAuto(false);
    setOpenManual(true);
  };
  const handleCloseManual = () => {
    setOpenManual(false);
    setOpenSelect(true);
  };

  return (
    <Container>
      <Typography variant="h1" fontSize="2.25rem" fontWeight="bold">
        Address Book
      </Typography>
      <Typography fontSize="1.125rem" sx={{ marginBottom: "24px" }}>
        Welcome back to your address book.
      </Typography>

      <AddressInputWrapper>
        <Typography variant="h3" fontSize="1.5rem" fontWeight="bold">
          Address:
        </Typography>
        <Grid
          container
          display="flex"
          justifyContent="space-between"
          marginTop="24px"
        >
          <Grid item md={8} xs={12}>
            <BequestInput value={concatAddress(currentAddr)} disabled />
          </Grid>
          <Grid item md={3} xs={12}>
            <BequestButton onClick={handleOpenSelect}>
              Select Address
            </BequestButton>
          </Grid>
        </Grid>
      </AddressInputWrapper>

      <SelectAddressDialog
        open={openSelect}
        handleClose={handleCloseSelect}
        handleOpenNext={handleOpenAuto}
        title="Select Address"
      />

      <AutoAddDialog
        open={openAuto}
        handleClose={handleCloseAuto}
        handleOpenNext={handleOpenManual}
      />

      <ManualAddDialog
        open={openManual}
        handleClose={handleCloseManual}
        handleOpenNext={handleOpenAuto}
        title="Add address"
      />
    </Container>
  );
};

export default AddressBook;
