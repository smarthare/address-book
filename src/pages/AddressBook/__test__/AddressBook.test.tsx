import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import AddressBook from "../../AddressBook";

afterEach(() => {
  cleanup();
});

test("Should render AddressBook component", () => {
  render(<AddressBook />);

  const pageElement = screen.getByTestId("addressBookPage");

  expect(pageElement).toBeInTheDocument();
  expect(screen.getByTestId("pageTitle")).toHaveTextContent("Address Book");
  expect(screen.getByTestId("pageInput")).toHaveValue("");
  expect(screen.getByTestId("pageBtn")).toHaveTextContent("Select Address");
});

test("AddressBook matches snapshot", () => {
  const component = render(<AddressBook />);

  expect(component).toMatchSnapshot();
});

test("Should display current address", () => {
  // const {}
  render(<AddressBook />);

})