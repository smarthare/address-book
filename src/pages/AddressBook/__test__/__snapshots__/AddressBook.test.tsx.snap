// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBook matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-qhvm03"
        data-testid="addressBookPage"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-11g9nu0-MuiTypography-root"
          data-testid="pageTitle"
        >
          Address Book
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-m94hbq-MuiTypography-root"
        >
          Welcome back to your address book.
        </p>
        <div
          class="css-1bxarir"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 css-11vjk9y-MuiTypography-root"
          >
            Address:
          </h3>
          <div
            class="MuiGrid-root MuiGrid-container css-vri7p7-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
            >
              <input
                class="css-sghe0z"
                data-testid="pageInput"
                disabled=""
                value=""
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
            >
              <button
                class="MuiButtonBase-root css-17jrrdy-MuiButtonBase-root"
                data-testid="pageBtn"
                tabindex="0"
                type="button"
              >
                Select Address
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-qhvm03"
      data-testid="addressBookPage"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-11g9nu0-MuiTypography-root"
        data-testid="pageTitle"
      >
        Address Book
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-m94hbq-MuiTypography-root"
      >
        Welcome back to your address book.
      </p>
      <div
        class="css-1bxarir"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-11vjk9y-MuiTypography-root"
        >
          Address:
        </h3>
        <div
          class="MuiGrid-root MuiGrid-container css-vri7p7-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-9drkqd-MuiGrid-root"
          >
            <input
              class="css-sghe0z"
              data-testid="pageInput"
              disabled=""
              value=""
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-3 css-iv7vbz-MuiGrid-root"
          >
            <button
              class="MuiButtonBase-root css-17jrrdy-MuiButtonBase-root"
              data-testid="pageBtn"
              tabindex="0"
              type="button"
            >
              Select Address
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
